src/Adapters/EventSource.Net46/EventSourceTelemetryModule.cs

```diff
@@ -0,0 +1,151 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="EventSourceTelemetryModule.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2017
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.Implementation;
+
+using Microsoft.ApplicationInsights.EventSourceListener.Implementation;
+
+namespace Microsoft.ApplicationInsights.EventSourceListener
+{
+    /// <summary>
+    /// A module to trace data submitted via .NET framework <seealso cref="System.Diagnostics.Tracing.EventSource" /> class.
+    /// </summary>
+    public class EventSourceTelemetryModule : EventListener, ITelemetryModule
+    {
+        private TelemetryClient client;
+        private bool constructed;  // Relying on the fact that constructed will be false initially (.NET default)
+        private List<EventSource> eventSourcesPresentAtConstruction;
+
+        /// <summary>
+        /// Gets or sets the list of EventSource listening requests (information about which EventSources should be traced).
+        /// </summary>
+        public IList<EventSourceListeningRequest> Sources { get; private set; }
```

Comment:
It will be cool to allow collection modification in runtime. Would you pls create an issue for this?